pkg/gcp/actuator/actuator.go

```diff
@@ -78,16 +79,23 @@ type Actuator struct {
 	ProjectName      string
 	Client           client.Client
 	RootCredClient   client.Client
-	GCPClientBuilder func(string, []byte) (ccgcp.Client, error)
+	GCPClientBuilder func(string, []byte, []configv1.GCPServiceEndpoint) (ccgcp.Client, error)
+	GCPEndpoints     []configv1.GCPServiceEndpoint
 }
 
 // NewActuator initializes and returns a new Actuator for GCP.
 func NewActuator(c, rootCredClient client.Client, projectName string) (*Actuator, error) {
+	endpoints, err := gcputils.GetServiceEndpoints(c)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
I expect this would be failing similar to the secretannotator controller if we weren't failing on that controller.